src/Leaves/FilterTabs.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Rhubarb\Leaf\Tabs\Leaves;
+
+use Rhubarb\Leaf\Leaves\Leaf;
+
+class FilterTabs extends Tabs
+{
+    protected function populateFilter()
+    {
+        $tab = $this->model->tabs[$this->model->selectedTab];
+        if ($tab instanceof FilterTabDefinition) {
+            return $tab->getFilter();
+        }
+
+        return null;
+    }
+
+    protected function bindEvents(Leaf $presenter)
+    {
+        if (property_exists($presenter, "getFilterEvent")) {
```

Comment:
you could write this `isset($presenter->getFilterEvent) && $presenter->getFilterEvent instanceof Event` as that actually acts as a type hint for IDEs and check that the property is set to an event. As it is, you're trying to use the property as an `Event` so long as the class has a property (which is `true` if the property is uninitialised so you could be doing this on `null`)